apps/feelslike/feels_like.star

```diff
@@ -0,0 +1,84 @@
+"""
+Applet: Feels Like
+Summary: Shows "feels like" weather
+Description: Displays the "feels like" temperature for the set location.
+Author: skalum
+"""
+
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+
+DEFAULT_LOCATION = """
+{
+	"lat": "37.7639316",
+	"lng": "-122.4030647",
+	"description": "451 Kansas St, San Francisco, CA 94107, USA",
+	"locality": "San Francisco",
+	"place_id": "ChIJSbhWnuF_j4ARbk1kioTNmEE",
+	"timezone": "America/Los_Angeles"
+}
+"""
+
+DEFAULT_API_KEY = "noKey"
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "apiKey",
+                name = "API key",
+                desc = "OpenWeather API key",
+                icon = "key",
+            ),
+            schema.Location(
+                id = "location",
+                name = "Location",
+                desc = "Location for which to display weather",
+                icon = "locationDot",
+            ),
+        ],
+    )
+
+def main(config):
+    apiKey = config.get("apiKey", DEFAULT_API_KEY)
+
+    location = json.decode(DEFAULT_LOCATION)
+    lat, lng = float(location["lat"]), float(location["lng"])
+
+    url = "https://api.openweathermap.org/data/3.0/onecall?lat={}&lon={}&appid={}&units=imperial".format(lat, lng, apiKey)
+    weather = http.get(url).json()
```

Comment:
I think it would be good to add some caching for these http calls. Even if the user provides their own API key, it'd be unfortunate if all their request quota gets eaten up faster that it has to.

Reply:
Agreed, I just know that weather can change very quickly... I will check how often they update their data.

Also, I didn't necessarily need this to be a public app, just made it for my wife who requested it! There's no way to create a private app, correct? So you don't need to spend time approving my prs?

Reply:
We can unfortunately not host private apps right now, but that might change in the future. There's of course always the option of self-hosting and pushing graphics to your Tidbyt [via the API](https://tidbyt.dev/docs/integrate/pushing-apps).